Source/Extensions/Blazorise.RichTextEdit/RichTextEdit.razor.cs

```diff
@@ -0,0 +1,194 @@
+            {
+                JSRuntime.InvokeVoidAsync( "blazoriseRichTextEdit.destroy", editorRef );
+                cleanup.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Called when [first after render asynchronous].
+        /// </summary>
+        protected override async Task OnFirstAfterRenderAsync()
+        {
+            var dotNetRef = DotNetObjectReference
+                .Create( this )
+                .DisposeWith( cleanup );
+
+            await JSRuntime.InvokeVoidAsync( "blazoriseRichTextEdit.initialize",
+                dotNetRef,
+                editorRef,
+                ToolbarContent != null ? toolbarRef : default,
+                ReadOnly,
+                PlaceHolder,
+                Theme == RichTextEditTheme.Snow ? "snow" : "bubble",
+                nameof( OnContentChanged ),
+                SubmitOnEnter,
+                nameof( OnEnter ) );
+
+            initialized = true;
+
+            if ( EditorContent != null )
+            {
+                var initialContent = await GetContentAsync();
+                await OnContentChanged( initialContent );
+            }
+        }
+
+        /// <summary>
+        /// Sets the editor content asynchronous.
+        /// </summary>
+        public ValueTask SetContentAsync( string html )
+            => JSRuntime.InvokeVoidAsync( "blazoriseRichTextEdit.setContent", editorRef, html );
+
+        /// <summary>
+        /// Gets the editor content asynchronous.
+        /// </summary>
+        public ValueTask<string> GetContentAsync()
+            => JSRuntime.InvokeAsync<string>( "blazoriseRichTextEdit.getContent", editorRef );
+
+        /// <summary>
+        /// Clears the editor content asynchronous.
+        /// </summary>
+        public ValueTask ClearAsync()
+            => JSRuntime.InvokeVoidAsync( "blazoriseRichTextEdit.clearContent", editorRef );
+
+        /// <summary>
+        /// Javascript callback for when content changes.
+        /// </summary>
+        [JSInvokable]
+        public Task OnContentChanged( string html )
+            => ContentChanged.InvokeAsync( html );
+
+        /// <summary>
+        /// Javascript callback for when enter is pressed.
+        /// </summary>
+        [JSInvokable]
+        public Task OnEnter()
+            => EnterPressed.InvokeAsync( true );
+
+        /// <summary>
+        /// Toggles the readonly state
+        /// </summary>
+        private async void SetReadOnly( bool value )
+        {
+            if ( initialized )
+            {
+                await JSRuntime.InvokeVoidAsync( "blazoriseRichTextEdit.setReadOnly", editorRef, value );
+            }
+        }
+
+        #endregion
+
+        #region Properties
+
+        [Inject] private IJSRuntime JSRuntime { get; set; }
+
+        /// <summary>
+        /// [Optional] Gets or sets the content of the toolbar.
+        /// </summary>
+        [Parameter] public RenderFragment ToolbarContent { get; set; }
```

Comment:
What do you think of renaming this to just Toolbar?

Reply:
fine